x/ecocredit/base/types/v1/msg_update_batch_metadata.go

```diff
@@ -0,0 +1,51 @@
+package v1
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/x/auth/migrations/legacytx"
+
+	"github.com/regen-network/regen-ledger/x/ecocredit"
+	"github.com/regen-network/regen-ledger/x/ecocredit/base"
+)
+
+var _ legacytx.LegacyMsg = &MsgUpdateBatchMetadata{}
+
+// Route implements the LegacyMsg interface.
+func (m MsgUpdateBatchMetadata) Route() string { return sdk.MsgTypeURL(&m) }
+
+// Type implements the LegacyMsg interface.
+func (m MsgUpdateBatchMetadata) Type() string { return sdk.MsgTypeURL(&m) }
+
+// GetSignBytes implements the LegacyMsg interface.
+func (m MsgUpdateBatchMetadata) GetSignBytes() []byte {
+	return sdk.MustSortJSON(ModuleCdc.MustMarshalJSON(&m))
+}
+
+// ValidateBasic does a sanity check on the provided data.
+func (m *MsgUpdateBatchMetadata) ValidateBasic() error {
+	if _, err := sdk.AccAddressFromBech32(m.Issuer); err != nil {
+		return sdkerrors.ErrInvalidAddress.Wrapf("issuer: %s", err)
+	}
+
+	if err := base.ValidateBatchDenom(m.BatchDenom); err != nil {
+		return sdkerrors.ErrInvalidRequest.Wrapf("batch denom: %s", err)
+	}
+
+	// we allow removing metadata for class and project but not for batch
```

Comment:
not related to this task but it seems like we can create a batch initially without any metadata (see `MsgCreateBatch`.`ValidateBasic()`) should this be fixed?

Reply:
I'll open a followup since this has already been approved.

Reply:
#1630